pandas/core/nanops.py

```diff
@@ -367,6 +367,32 @@ def _wrap_results(result, dtype: np.dtype, fill_value=None):
     return result
 
 
+def _datetimelike_compat(func):
```

Comment:
FYI i don't think this needs to be private, but nbd.

Reply:
also as a follow up need to ensure decorator preserves types #33455

Reply:
will do follow-up to annotate new_func.  can we start typing mask as `np.ndarray[bool]` or something like that?

Reply:
The important thing is that the decoratored function signature is preserved. It may not be so important, in the grand scheme of things, to type check this function. though can do as well.